src/frequenz/channels/_receiver.py

```diff
@@ -155,11 +155,14 @@
 
 from abc import ABC, abstractmethod
 from collections.abc import Callable
-from typing import Generic, Self
+from typing import TYPE_CHECKING, Any, Generic, Self, TypeGuard
 
 from ._exceptions import Error
 from ._generic import MappedMessageT_co, ReceiverMessageT_co
 
+if TYPE_CHECKING:
+    from ._select import Selected
```

Comment:
:unamused: 

Oh, this is also an ugly side-effect of moving part of the implementation of `select` to `Receiver`. We could consider moving `Selected` to this file, if we are mixing stuff up anyway.

Reply:
True, maybe also rename it so it sounds like `WrappedMessage` or `ReceivedMessage`, and it would be decoupled from `select` completely?

If not, it feels a bit weird, defining a `select` feature in the receiver module.

Reply:
I don't know, for me it is unfortunate that we need to implement part of `select()` in receiver but it doesn't make sense to use weirder names only to try to hide that. `Selected` / `WrappedMessage` / `ReceivedMessage` will only be used by select anyways, I would keep calling it `Selected`.

I really have double feelings about this PR, for one side I think it improves the UX slightly, but I'm not sure if enough to worth a deprecation and pushing people to update their code, and for the other side I think it makes the implementation slightly less clean (more coupling between `Receiver` and `select`), but I'm not sure if enough to completely oppose to the change :smile: :disappointed:

Reply:
> I think it makes the implementation slightly less clean (more coupling between `Receiver` and `select`)

I would look at it as an increased commitment to having `select` as a core component of the channels package, select is very deeply related to receivers.

And if it was a different language, we wouldn't even notice the coupling, because no language makes so much noise about these things as python.

Reply:
> I think it improves the UX slightly,

I think it is more than a slight improvement for new users, because they don't have to import the weird-looking `selected_from` separately, so there's less confusion.

Reply:
I think we can support the the old way for long to stay backwards compatible to avoid annoying users though?

Reply:
Oh right, so we can't rename `Selected`.

Also you mean add the new stuff, but don't deprecate the `selected_from` method?

Reply:
Yeah, maybe we can deprecate it after a few more releases, when the new form is more widely used, to confirm it is really better than the old form.

Reply:
> Oh right, so we can't rename `Selected`.

We can always rename it and create a (deprecated) alias using the old name. But personally I wouldn't do it, I think that name is fine.

Anyway, as I said, my main complain about this PR is the current `is_selected()` name, if we get a better name I'm OK with merging. I think my preferred name is `is_source_of()` but I can also live with `matches()`

Reply:
About names, I must admit I also thought about `was`, which sucks as a name taken out of context, but reads very nicely when put in `receiver.was(selected)` :smile: :grimacing: 
ChatGPT also suggested `receiver.is_active(selected)` which doesn't reads as nicely as `receiver.is_actively(selected)` but makes much more sense as a function name.

Reply:
Similar to `was`, but maybe a bit more obvious, and independent of the variable naming conventions for `selected`, maybe `receiver.produced(selected)`.  `was` doesn't work IMO, because it depends on the variable name `selected`, which we don't have any control over.

But I'm still in favour of `matches`, because it has the coolest sound, with a lispy/functional feel, in a select block:

```python
async for selected in select(power_rx, timer):
    if power_rx.matches(selected):
        print(f"Power: {power}")
    elif timer.matches(selected):
        print("Timer triggered")
```

Reply:
In the line of `produced`, what about `triggered`?

```py
async for selected in select(power_rx, timer):
    if power_rx.triggered(selected):
        print(f"Power: {power}")
    elif timer.triggered(selected):
        print("Timer triggered")
```

What it bothers me about `matches` is it is really the `selected` that matches the receiver, not the other way around (if we could use `selected.matches(receiver)` that would be a :100: ). But as I said, I can live with `matches`.

Reply:
You were right.  I'm also having doubts about `matches`.  I will update the PR to say `triggered`.

Reply:
:scream: